designs/nutanix-datacenterconfig-credentialref.md

```diff
@@ -0,0 +1,51 @@
+# Nutanix Datacenter Config Credential Reference
+
+## Introduction
+
+**Problem:** When a management cluster and a workload cluster are present on two different Nutanix Prism Central instances,
+using same credentials to authenticate to both instances is not possible. To remedy this, we need to associate credentials
+with corresponding Nutanix Datacenter Configs.
+
+## Overview of Solution
+With this feature, a user can specify credentials reference (i.e. Name and Namespace of the secret containing the credentials)
+in the Nutanix Datacenter Config. This info will be fetched and used to authenticate to the Nutanix Prism Central instance
+when creating the workload cluster.
+
+### Solution Details
+Extend the `NutanixDatacenterConfig` CRD to include a new field `credentialRef` storing name, namespace, and kind (similar to how
+`credentialsRef` is defined in the `NutanixCluster` CRD).
+
+```go
+// NutanixDatacenterConfigSpec defines the desired state of NutanixDatacenterConfig.
+type NutanixDatacenterConfigSpec struct {
+    ...
+    // CredentialRef is the reference to the secret that contains the credentials
+    // for the Nutanix Prism Central
+    // +optional
+    CredentialRef *NutanixCredentialReference `json:"credentialRef,omitempty"`
```

Comment:
Since this is optional, what should be the behavior when not present? I'm assuming defaulting to a secret with a fixed name in a fixed namespace?

Reply:
I see that this is explained bellow. Could we add that piece of info here as well?

Reply:
Sure, we can add it here. @jiayiwang7 looking at Ref, lack of Namespace is a potential concern.

Reply:
do we need the namespace? We mainly operate on objects/CRDs that are in the `eksa-system` namespace. This also prevents eks-a controller from needing access to secrets in all namespaces and limits the access it needs. 
From EKS-A side, we can ask users to create the secret in the `eksa-system` ns.

Reply:
If we are fine with the secrets always being created in `eksa-system` namespace, we can definitely use the `Ref` type.